src/pages/Facility/settings/organizations/components/EditFacilityOrganizationSheet.tsx

```diff
@@ -0,0 +1,168 @@
+
+  const { mutate: updateOrganization, isPending } = useMutation({
+    mutationFn: (body: FacilityOrganizationEdit) =>
+      mutate(facilityOrganizationApi.update, {
+        pathParams: { facilityId, organizationId: organization.id },
+        body,
+      })(body),
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: ["facilityOrganization", "list", facilityId],
+      });
+      toast.success(t("organization_updated_successfully"));
+      setOpen(false);
+    },
+    onError: (error) => {
+      const errorData = error.cause as { errors: { msg: string }[] };
+      errorData.errors.forEach((er) => {
+        toast.error(er.msg);
+      });
+    },
+  });
+
+  const handleSubmit = () => {
+    if (!name.trim()) {
+      toast.error(t("please_enter_organization_name"));
+      return;
+    }
+
+    updateOrganization({
+      name: name.trim(),
+      description: description.trim() || undefined,
+      org_type: orgType,
+      facility: facilityId,
+      parent: organization?.parent?.id,
+    });
+  };
+
+  return (
+    <Sheet open={open} onOpenChange={setOpen}>
+      <SheetTrigger asChild>
+        <Button variant="white" size={isMobile ? "xs" : "sm"}>
+          <PenLine className="h-4 w-4" />
+          <span className="hidden lg:inline">{t("edit")}</span>
+        </Button>
+      </SheetTrigger>
+      <SheetContent>
+        <SheetHeader>
+          <SheetTitle>{t("edit_department_team")}</SheetTitle>
+          <SheetDescription>
+            {t("edit_department_team_description")}
+          </SheetDescription>
+        </SheetHeader>
+        <div className="space-y-6 py-4">
+          <div className="space-y-2">
+            <label className="text-sm font-medium">{t("name")}</label>
+            <Input
+              value={name}
+              onChange={(e) => setName(e.target.value)}
+              placeholder={t("enter_department_team_name")}
+            />
+          </div>
+
+          <div className="space-y-2">
+            <label className="text-sm font-medium">{t(`type`)}</label>
+            <Select
+              value={orgType}
+              onValueChange={(value: OrgType) => setOrgType(value)}
+            >
+              <SelectTrigger>
+                <SelectValue placeholder={t("select_organization_type")} />
+              </SelectTrigger>
+              <SelectContent>
+                {ORG_TYPES.map((type) => (
+                  <SelectItem key={type.value} value={type.value}>
+                    {type.label}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+          </div>
+
+          <div className="space-y-2">
+            <label className="text-sm font-medium">{t("description")}</label>
+            <Textarea
+              value={description}
+              onChange={(e) => setDescription(e.target.value)}
+              placeholder={t("enter_department_team_description")}
+            />
+          </div>
```

Comment:
let use form, formfield and react hook form to maintain the form state and validation, which makes the implementation clean and makes showing error message easy